.github/actions/createOrUpdateStagingDeploy/index.js

```diff
@@ -195,14 +195,31 @@ const {execSync} = __nccwpck_require__(3129);
  * @returns {Array}
  */
 function getPullRequestsMergedBetween(fromRef, toRef) {
-    const command = `git log --format="%s" ${fromRef}...${toRef}`;
+    const command = `git log --format="{[%B]}" ${fromRef}...${toRef}`;
     console.log('Getting pull requests merged between the following refs:', fromRef, toRef);
     console.log('Running command: ', command);
     const localGitLogs = execSync(command).toString();
-    return _.map(
-        [...localGitLogs.matchAll(/Merge pull request #(\d{1,6}) from (?!Expensify\/(?:master|main|version-))/g)],
+
+    // Parse the git log into an array of commit messages between the two refs
+    const commitMessages = _.map(
+        [...localGitLogs.matchAll(/{\[([\s\S]*?)\]}/gm)],
         match => match[1],
     );
+    console.log(`A list of commits made between ${fromRef} and ${toRef}:\n${commitMessages}`);
+
+    // We need to find which commit messages correspond to merge commits and get PR numbers.
+    // Additionally, we omit merge commits made while cherry picking using negative lookahead in the regexp.
+    const pullRequestIDs = _.reduce(commitMessages, (mergedPRs, commitMessage) => {
+        const mergeCommits = [...commitMessage.matchAll(/Merge pull request #(\d{1,6}) from (?!(?:Expensify\/(?:master|main|version-))|(?:([\s\S]*?)\(cherry picked from commit .*\)\s*))/gm)];
+
+        // Get the PR number of the first match (there should not be multiple matches in one commit message)
+        if (_.size(mergeCommits)) {
+            mergedPRs.push(mergeCommits[0][1]);
+        }
+        return mergedPRs;
+    }, []);
```

Comment:
And then I decided to use reduce on the list of commit messages. I use the regexp we have developed (slightly adjusted) to match the PR numbers. Then if some match found, I push the group to the accumulator array.